packages/adapter/src/lib/adapter/adapter.ts

```diff
@@ -3203,8 +3203,8 @@ export class AdapterClass extends EventEmitter {
         const { options, callback, obj } = _options;
         let { id } = _options;
 
-        obj.from = obj.from || 'system.adapter.' + this.namespace;
-        obj.user = obj.user || (options ? options.user : '') || SYSTEM_ADMIN_USER;
+        obj.from = obj.from || `system.adapter.${this.namespace}`;
+        obj.user = obj.user || (options && options.user) || SYSTEM_ADMIN_USER;
```

Comment:
```suggestion
        obj.user = obj.user || options?.user || SYSTEM_ADMIN_USER;
```

Reply:
These are typescript tricks :)

Reply:
lol yeah I have commited it for you we can merge